### PR TITLE
Z velocity to position

### PR DESCRIPTION
**Pull request by aftersomemath**
@MiscM Started this and I wrapped up a few details.

Completely breaks functional compatibility but not software compatibility due to significant restructuring of HLM that is required. I plan on doing this restructuring as part of senior design so it can't be in this PR. The restructure will achieve linear profile interpolation (necessary for senior design) with a revamp of HLM that allows the intuitive nature of velocity and position targets to be used while still causing completely filled motion points to be output.

Technically the software is still correct, its just that LLM won't ever takeoff since the QuadVelocity controller looks at Z position which isn't being filled out by HLM.

### CODE REVIEW

**Comment by amiller27** [comment]
So I'm thinking we wait to merge this until the rest of it is ready? Is that good with you?

**Comment by aftersomemath** [comment]
Yes but please review so that we can fix things sooner rather than later.

File: scripts/waypoints_test.py [hunk 1]
```diff
@@ -22,7 +21,7 @@ def constrain(x, l, h):
     safety_client = SafetyClient('motion_command_coordinator')
     assert safety_client.form_bond()
 
-    velocity_pub = rospy.Publisher('movement_velocity_targets', TwistStampedArray, queue_size=0)
+    motion_point_pub = rospy.Publisher('motion_point_targets', MotionPointStampedArray, queue_size=0)
```
Comment (Riteshlogin) [inline scripts/waypoints_test.py]: What subscribes to this and are there any other PR's that this depends on?

File: scripts/waypoints_test.py [hunk 2]
```diff
@@ -40,15 +39,20 @@ def constrain(x, l, h):
 
     # Target points in global (X, Y, Z) coordinates
```
Comment (Riteshlogin) [inline scripts/waypoints_test.py]: This comment is not entirely accurate

File: scripts/waypoints_test.py (hunk 1 above)
Comment (aftersomemath) [reply scripts/waypoints_test.py]: Low level motion. Nothing depends on this but the whole PR depends on one that doesnt exist yet.

**Comment by asaba96** [comment]
What kind of changes to HLM were you envisioning? Supporting motion points commands? Revamping the takeoff/land tasks?

**Comment by asaba96** [comment]
Also please no breaking changes being merged. Otherwise you will force us to use an old commit version of master to do our work.

File: scripts/waypoints_test.py (hunk 1 above)
Comment (asaba96) [reply scripts/waypoints_test.py]: What does this pr depend on? Another topic or node? I thought you made a motion point transcriber node.

Comment (aftersomemath) [reply scripts/waypoints_test.py]: It depends on a hlm outputting motion points with z position filled out.

File: scripts/waypoints_test.py [hunk 3]
```diff
@@ -90,53 +94,43 @@ def constrain(x, l, h):
...
-            if abs(yaw_difference) < 0.15:
-                target = next(waypoints_iter, target)
+        motion_point.motion_point.pose.position.z = target[2]
+
+        motion_point_msg = MotionPointStampedArray()
+        motion_point_msg.motion_points = [motion_point]
+        motion_point_pub.publish(motion_point_msg)
+
+        #Sets next target after a set time has passed
+        if ((rospy.Time.now().to_sec() - time) > 1):
```
[the remark this answers is not in the document]
Comment (aftersomemath) [reply scripts/waypoints_test.py]: It is never hit. This script just tests controller performance in a very raw manner.

File: scripts/waypoints_test.py (hunk 1 above)
Comment (asaba96) [reply scripts/waypoints_test.py]: Velocity commands require a z position now?

**Comment by aftersomemath** [comment]
HLM needs to maintain a simple way to tasks to assign positions and velocities but needs to provide full motion point plans at all times. I plan to add a simple profile generator that generates plans from target positions or velocities.

File: scripts/waypoints_test.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Riteshlogin) [reply scripts/waypoints_test.py]: Hmm, ok. I think that by competition time, this should be replaced by logic that checks the status of the goal or the difference between the target and the current position instead.

[the remark this answers is not in the document]
Comment (Riteshlogin) [reply scripts/waypoints_test.py]: Never mind

File: src/LandPlanner.cpp [hunk 4]
```diff
@@ -100,28 +98,13 @@ bool LandPlanner::getTargetTwist(const ros::Time& time,
...
     }
 
-    if(state_ == LandState::ACCELERATE_TO_DESCENT_VELOCITY) {
-        requested_z_vel_ = std::max(requested_z_vel_ - ((time - last_update_time_).toSec() * descend_acceleration_), descend_rate_);
-        if(requested_z_vel_ <= descend_rate_) {
-            state_ = LandState::DESCEND;
-        }
-    }
-    else if(state_ == LandState::DESCEND)
+    if(state_ == LandState::DESCEND)
```
Comment (amiller27) [inline src/LandPlanner.cpp]: Why are we not ramping the descent rate anymore?

Comment (aftersomemath) [reply src/LandPlanner.cpp]: We are still ramping the setpoint, its just that I didn't want to handle ramping 2 derivatives down. In theory this should be replaced by a plan generator in HLM. Which may happen. This gave acceptable performance in the sim.

File: scripts/waypoints_test.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (aftersomemath) [reply scripts/waypoints_test.py]: Fixed

File: scripts/waypoints_test.py (hunk 2 above)
Comment (aftersomemath) [reply scripts/waypoints_test.py]: Fixed

File: src/LandPlanner.cpp (hunk 4 above)
Comment (aftersomemath) [reply src/LandPlanner.cpp]: Done

**Pull request merged**